base/loading.jl

```diff
@@ -263,6 +263,9 @@ const _require_dependencies = Any[] # a list of (path, mtime) tuples that are th
 const _track_dependencies = Ref(false) # set this to true to track the list of file dependencies
 function _include_dependency(_path::AbstractString)
     prev = source_path(nothing)
+    if myid() != 1 && prev === nothing
+        prev = remotecall_fetch(abspath, 1, ".")
+    end
```

Comment:
it seems like this may confuse the SOURCE_PATH pop in `include_from_node1`.

I think it'll be clearer to be explicit here:
```julia
    prev = source_path(nothing)
    if prev === nothing
        if myid() == 1
            path = abspath(_path)
        else
            path = joinpath(remotecall_fetch(abspath, 1, "."), _path)
        end
    else
        path = joinpath(dirname(prev), _path)
    end
    if myid() == 1 && _track_dependencies[]
        push!(_require_dependencies, (path, mtime(path)))
    end
```